sdcflows/workflows/topup.py

```diff
@@ -0,0 +1,359 @@
+        used). The workflow raises a ``ValueError`` when no images with
+        opposed PE blips are found.
+    epi_pe_dir : str
+        The baseline PE direction.
+    epi_trt: float
+        Total readout time of the EPI (should be identical to TRT of fmaps_epi)
+    matched_pe_dir : str
+        Phase encoding direction of matching fmap EPI
+    opposed_pe_dir : str
+        Phase encoding direction of opposed fmap EPI
+    in_reference : pathlike
+        The baseline reference image (must correspond to ``epi_pe_dir``).
+
+    Outputs
+    -------
+    fieldmap : pathlike
+        Topup estimated fieldmap (Hz)
+    magnitude : pathlike
+        The ``fmaps_epi`` after unwarping
+    """
+
+    workflow = Workflow(name=name)
+    workflow.__desc__ = """\
+A B0-nonuniformity map (or *fieldmap*) was estimated based on two (or more)
+echo-planar imaging (EPI) references with opposing phase-encoding
+directions, with `Topup`).
+"""
+
+    inputnode = pe.Node(niu.IdentityInterface(
+        fields=['fmaps_epi', 'epi_pe_dir', 'epi_trt', 'in_reference', 'matched_pe_dir',
+                'opposed_pe_dir']),
+        name='inputnode')
+
+    outputnode = pe.Node(niu.IdentityInterface(
+        fields=['fieldmap', 'magnitude']),
+        name='outputnode')
+
+    prepare_epi_wf = init_prepare_epi_wf(omp_nthreads=omp_nthreads,
+                                         matched_pe=matched_pe,
+                                         name="prepare_epi_wf")
+
+    epi_merge2list = pe.Node(niu.Merge(2), name='epi_merge2list')
+    pedirs_merge2list = pe.Node(niu.Merge(2), name='pedir_merge2list')
+    trt_merge2list = pe.Node(niu.Merge(2), name='trt_merge2list')
+
+    merger = pe.Node(fsl.Merge(dimension='t'), name='merger')
+
+    topup = pe.Node(fsl.TOPUP(), name='topup')
+
+    workflow.connect([
+        (inputnode, prepare_epi_wf, [
+            ('fmaps_epi', 'inputnode.maps_pe'),
+            ('epi_pe_dir', 'inputnode.epi_pe'),
+            ('in_reference', 'inputnode.ref')]),
+        (prepare_epi_wf, epi_merge2list, [
+            ('outputnode.matched_pe', 'in1'),
+            ('outputnode.opposed_pe', 'in2')]),
+        (epi_merge2list, merger, [
+            ('out', 'in_files')]),
+        (merger, topup, [
+            ('merged_file', 'in_file')]),
+        (inputnode, pedirs_merge2list, [
+            (('matched_pe_dir', _get_pedir_topup), 'in1'),
+            (('opposed_pe_dir', _get_pedir_topup), 'in2')]),
+        (pedirs_merge2list, topup, [
+            ('out', 'encoding_direction')]),
+        (inputnode, trt_merge2list, [
+            ('epi_trt', 'in1'),
+            ('epi_trt', 'in2')]),
+        (trt_merge2list, topup, [
+            ('out', 'readout_times')]),
+        (topup, outputnode, [
+            ('out_field', 'fieldmap'),
+            ('out_corrected', 'magnitude')]),
+    ])
+
+    return workflow
+
+
+def init_prepare_epi_wf(omp_nthreads, matched_pe=False,
```

Comment:
Can you unpack for me how much of this workflow overlaps the existing analog?

Reply:
In #106 all edits were done in `pepolar.py`, so should be easier to evaluate the overlap